storey/dtypes.py

```diff
@@ -46,6 +44,53 @@ def get_window_start_time(self):
         return datetime.now().timestamp() * 1000
 
 
+class WindowsBase:
+    def __init__(self, period, max_window, smallest_window, windows_str):
+        self.max_window_millis = max_window
+        self.smallest_window_millis = smallest_window
+        self.period_millis = period
+        self.windows_str = windows_str
+        self.windows_millis = [parse_duration(win) for win in windows_str]
+        self.total_number_of_buckets = int(self.max_window_millis / self.period_millis)
+
+
+class FixedWindows(WindowsBase):
+    def __init__(self, windows):
+        max_window_millis = parse_duration(windows[-1])
+        smallest_window_millis = parse_duration(windows[0])
+        WindowsBase.__init__(self, smallest_window_millis / bucketPerWindow,
```

Comment:
I don't think `smallest_window_millis / bucketPerWindow` is guaranteed to be an integer.

Reply:
well it's in milliseconds and `bucketPerWindow` is 10, so it should that's fine. the windows aren't gonna be in milliseconds